File: product-service/src/main/java/com/intexsoft/webshop/productservice/repository/CategoryRepository.java
package com.intexsoft.webshop.productservice.repository;

import com.intexsoft.webshop.productservice.model.Category;
import org.springframework.data.jpa.repository.JpaRepository;

import java.util.Optional;

public interface CategoryRepository extends JpaRepository<Category, Long> {
    Optional<Category> findByNameIgnoreCase(String name);
}

File: product-service-kt/src/main/kotlin/com/intexsoft/webshop/productservicekt/repository/CategoryRepository.kt
package com.intexsoft.webshop.productservicekt.repository

import com.intexsoft.webshop.productservicekt.model.Category
import org.springframework.data.jpa.repository.JpaRepository

interface CategoryRepository : JpaRepository<Category, Long> {
    fun findByNameIgnoreCase(name: String?): Category?
}
